tests/appium/project/src/test/kotlin/br/com/zup/beagle/cucumber/steps/ScrollViewScreenSteps.kt

```diff
@@ -16,135 +16,355 @@
-    malesuada ullamcorper dapibus donec aenean mattis odio mi nulla gravida. tellus metus imperdiet justo mattis 
-    eros sodales potenti nibh nisl tincidunt, metus etiam cubilia amet donec primis sapien erat dictumst. Accumsan 
-    etiam himenaeos tempor integer habitasse curae ac, tincidunt laoreet taciti nisl habitasse conubia, maecenas nec
-    velit vitae amet varius. scelerisque vel fringilla consequat justo curabitur nam massa vitae, tempus tempor 
-    sit torquent massa malesuada ullamcorper, laoreet elementum nam pharetra tempus nam mauris. sociosqu dictum 
-    malesuada lectus suscipit ullamcorper aliquet pulvinar semper laoreet, vulputate aliquam nibh odio donec ligula
-     bibendum suspendisse, facilisis ut lobortis lacus tortor hendrerit integer posuere. phasellus egestas dui hac 
-    auctor faucibus purus accumsan arcu, sem vivamus rhoncus pharetra aliquam ornare curabitur rutrum, ut venenatis
-     proin iaculis orci gravida molestie."""
+private const val NEW_TEXT_PREFIX = "Lorem ipsum diam luctus"
+private const val NEW_TEXT_SUFFIX = "proin iaculis orci gravida molestie."
 
 class ScrollViewScreenSteps : AbstractStep() {
     override var bffRelativeUrlPath = "/scrollview"
 
+    // cache
+    companion object {
+        var scrollViewElement1: MobileElement? = null
+        var scrollViewElement2: MobileElement? = null
+        var scrollViewElement3: MobileElement? = null
+    }
+
     @Before("@scrollview")
     fun setup() {
         loadBffScreen()
     }
 
     @Given("^that I'm on the scrollview screen$")
     fun checkScrollViewScreen() {
-        waitForElementWithTextToBeClickable(SCROLLVIEW_SCREEN_HEADER, false, false)
+        waitForElementWithTextToBeClickable(SCROLLVIEW_SCREEN_HEADER, likeSearch = false, ignoreCase = false)
     }
 
-    @When("^I have a horizontal scroll configured$")
-    fun checkHorizontalScrollText() {
-        waitForElementWithTextToBeClickable("Horizontal", false, false)
+    @When("^I access ScrollView (.*)$")
+    fun cacheScrollView(scrollViewElementNumber: Int) {
+        when (scrollViewElementNumber) {
+            1 -> {
+                scrollViewElement1 = getScrollViewElement(1)
+            }
+            2 -> {
+                scrollViewElement2 = getScrollViewElement(2)
+            }
+            else -> {
+                scrollViewElement3 = getScrollViewElement(3)
+            }
+        }
     }
 
-    @Then("^scrollview screen should perform the scroll action horizontally$")
-    fun validateHorizontalScroll() {
-        scrollDownToElementWithText("Horizontal", false, false)
+    @When("^I confirm that the ScrollView (.*) is not showing a button with text \"(.*)\" by default$")
+    fun confirmScrollView1IsNotShowingButton(scrollViewElementNumber: Int, buttonText: String) {
+        when (scrollViewElementNumber) {
+            1 -> {
+                Assert.assertFalse(isButtonShowingInsideOfScrollView(scrollViewElement1!!, buttonText))
```

Comment:
If you apply the above suggestion, you will no longer need to use the operator `!!`.
```suggestion
                Assert.assertFalse(isButtonShowingInsideOfScrollView(scrollViewElement1, buttonText))
```